arm-visualstudio/2014-04-01-preview/swagger/Csm.json

```diff
@@ -0,0 +1,818 @@
+        }
+      }
+    },
+    "AccountResourceListResult": {
+      "description": "The response to an account resource list GET request.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Array of resource details.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AccountResource"
+          }
+        }
+      }
+    },
+    "AccountResourceRequest": {
+      "description": "The body of a PUT request to modify a Visual Studio account resource.",
+      "type": "object",
+      "properties": {
+        "accountName": {
+          "description": "The account name.",
+          "type": "string"
+        },
+        "location": {
+          "description": "The Azure instance location.",
+          "type": "string"
+        },
+        "operationType": {
+          "description": "The type of the operation.",
+          "enum": [
+            "unknown",
+            "create",
+            "update",
+            "link"
+          ],
+          "x-ms-enum": {
+            "name": "AccountResourceRequestOperationType"
+          }
+        },
+        "properties": {
+          "description": "The custom properties of the resource.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "x-ms-client-flatten": true
+        },
+        "tags": {
+          "description": "The custom tags of the resource.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "CheckNameAvailabilityParameter": {
+      "description": "The body of a POST request to check name availability.",
+      "type": "object",
+      "properties": {
+        "resourceName": {
+          "description": "The name of the resource to check availability for.",
+          "type": "string"
+        },
+        "resourceType": {
+          "description": "The type of resource to check availability for.",
+          "type": "string"
+        }
+      }
+    },
+    "CheckNameAvailabilityResult": {
+      "description": "The response to a name availability request.",
+      "type": "object",
+      "properties": {
+        "message": {
+          "description": "The message describing the detailed reason.",
+          "type": "string"
+        },
+        "nameAvailable": {
+          "description": "The value which indicates whether the provided name is available.",
+          "type": "boolean"
+        }
+      }
+    },
+    "ExtensionResource": {
+      "description": "The response to an extension resource GET request.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/AccountResource"
+        }
+      ],
+      "properties": {
+        "plan": {
+          "description": "The extension plan that was purchased.",
+          "$ref": "#/definitions/ExtensionResourcePlan"
+        }
+      }
+    },
```

Comment:
Since extension resource support PATCH and PATCH requires all the properties to be optional we need different model for PATCH, something like


    "ExtensionResourceUpdateRequest": {
      "description": "The body of an extension resource PATCH request.",
      "type": "object",
      "properties": {
        "plan": {
          "description": "Extended information about the plan being purchased for this extension resource.",
          "$ref": "#/definitions/ExtensionResourcePlan"
        },
        "properties": {
          "description": "A dictionary of extended properties. This property is currently unused.",
          "type": "object",
          "additionalProperties": {
            "type": "string"
          },
          "x-ms-client-flatten": true
        },
        "tags": {
          "description": "A dictionary of user-defined tags to be stored with the extension resource.",
          "type": "object",
          "additionalProperties": {
            "type": "string"
          }
        }
    }
```